caikit_nlp/data_model/generation.py

```diff
@@ -73,6 +73,41 @@ class TuningConfig(DataObjectBase):
     # encoder_hidden_size: int # Optional -  The hidden size of the prompt encoder.
 
 
+@caikit.core.dataobject(package="caikit_data_model.caikit_nlp")
+class LoraTuningConfig(DataObjectBase):
+    # Lora attention dimension.
+    r: int
+    # The names of the modules to apply Lora to.
+    target_modules: Union[List[str], str]
+    # The alpha parameter for Lora scaling.
+    lora_alpha: int
+    # The dropout probability for Lora layers.
+    lora_dropout: float
+    # Set this to True if the layer to replace stores weight like (fan_in, fan_out).
+    # For example, gpt-2 uses Conv1D which stores weights like (fan_in, fan_out)
+    # and hence this should be set to True.
+    fan_in_fan_out: bool
+    # Bias type for Lora. Can be ‘none’, ‘all’ or ‘lora_only’.
+    # If ‘all’ or ‘lora_only’, the corresponding biases will be updated during training.
+    # Be aware that this means that, even when disabling the adapters,
+    # the model will not produce the same output
+    # as the base model would have without adaptation.
+    bias: str
+    # List of modules apart from LoRA layers to be set as trainable
+    # and saved in the final checkpoint.
+    modules_to_save: List[str]
+    # The layer indexes to transform, if this argument is specified,
+    # it will apply the LoRA transformations
+    # on the layer indexes that are specified in this list.
+    # If a single integer is passed,
+    # it will apply the LoRA transformations on the layer at this index.
+    layers_to_transform: Union[List[int], int]
+    # The layer pattern name, used only if layers_to_transform is different from None and
+    # if the layer pattern is not in the common layers pattern.
+    layers_pattern: str
+    output_model_types: List[str]
+
```

Comment:
I didnt add few  parameters from LoraConfig that are less frequently used. Everything that we will want to tweak is added